nestedListIterator.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * // This is the interface that allows for creating nested lists.
+ * // You should not implement it, or speculate about its implementation
+ * public interface NestedInteger {
+ *
+ *     // @return true if this NestedInteger holds a single integer, rather than a nested list.
+ *     public boolean isInteger();
+ *
+ *     // @return the single integer that this NestedInteger holds, if it holds a single integer
+ *     // Return null if this NestedInteger holds a nested list
+ *     public Integer getInteger();
+ *
+ *     // @return the nested list that this NestedInteger holds, if it holds a nested list
+ *     // Return null if this NestedInteger holds a single integer
+ *     public List<NestedInteger> getList();
+ * }
+ */
+ 
+public class NestedIterator implements Iterator<Integer> {
+    Stack<NestedInteger> st;
+    //In the constructor, we push all the nestedList into the stack from back to front
+    public NestedIterator(List<NestedInteger> nestedList) {
+        st = new Stack<>();
+        for(int i= nestedList.size()-1; i>=0 ; i--)
+            st.push(nestedList.get(i));
+    }
```

Comment:
This is an intermediate solution still. In stack in constructor ideally we should be having a single element. Here we have all elements. Then we are popping and checking if that is a nestedList or an integer. Try miming the ordinary iterator for Nested Iterator